### PR TITLE
Chore/fetch interceptor plat 236

### PR DESCRIPTION
**Pull request by rav3n11**
Add an http wrapper method to add an authorization header to each request sent to the backend

### CODE REVIEW

File: frontend/public/data/env.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+window.tabiyaConfig= {
+  "BACKEND_URL": window.btoa("https://dev.tabiya.tech/api")
+}
```
Comment (irumvanselme) [inline frontend/public/data/env.js:3]: Why Commit the file? and hen add it in the .gitignore

File: frontend/src/modelInfo/modelInfo.service.ts [hunk 2]
```diff
@@ -33,13 +34,7 @@ type ModelInfoTypeAPISpecs =
 export const UPDATE_INTERVAL = 10000; // In milliseconds
 
 export default class ModelInfoService {
-  readonly modelInfoEndpointUrl: string;
-  readonly apiServerUrl: string;
-
-  constructor(apiServerUrl: string) {
-    this.modelInfoEndpointUrl = `${apiServerUrl}/models`;
-    this.apiServerUrl = apiServerUrl;
-  }
```
Comment (irumvanselme) [inline frontend/src/modelInfo/modelInfo.service.ts:42]: Why did you change the constructor for the `ModelInfoService`. I thought the only change would be to change the param passed to the constructor instead of being hard-coded, we pass from the one constructed from the `env.json`.

Or, is there any specific reason?

File: .github/workflows/build-frontend.yml [hunk 3]
```diff
@@ -33,16 +33,17 @@ jobs:
...
-            echo "\"COGNITO_CLIENT_SECRET\":\"${{ inputs.cognito-client-secret }}\"," >> frontend/public/data/env.json
-            echo "}" >> frontend/public/data/env.json
-            cat frontend/public/data/env.json
+          echo "window.tabiyaConfig = {" > frontend/public/data/env.js
+          echo "\"BACKEND_URL\": btoa(\"${{ inputs.backend-url }}\")," >> frontend/public/data/env.js
+          echo "\"AUTH_URL\": btoa(\"${{ inputs.auth-url }}\")," >> frontend/public/data/env.js
+          echo "\"COGNITO_CLIENT_ID\": btoa(\"${{ inputs.cognito-client-id }}\")," >> frontend/public/data/env.js
+          echo "\"COGNITO_CLIENT_SECRET\": btoa(\"${{ inputs.cognito-client-secret }}\")" >> frontend/public/data/env.js
+          echo "};" >> frontend/public/data/env.js
+          cat frontend/public/data/env.js
```
Comment (irumvanselme) [inline .github/workflows/build-frontend.yml:46]: Looks good,

But once this is merged, I think we will need to add a section in frontend/read-me to specify how before running the frontend we need to first add this env.js in order to for the frontend to run correctly. we need to have a perfect env.js or in the root we add env.example.js and add a comment to move it to the public/data/env.js

**Review by irumvanselme — CHANGES_REQUESTED** [review]
added some comments

otherwise 

LGTM

File: frontend/src/modeldirectory/ModelDirectory.tsx [hunk 4]
```diff
@@ -26,9 +27,9 @@ export const DATA_TEST_ID = {
 export const SNACKBAR_ID = {
   INTERNET_ERROR: `internet-error-${uniqueId}`,
 };
-const importDirectorService = new ImportDirectorService("https://dev.tabiya.tech/api");
-const modelInfoService = new ModelInfoService("https://dev.tabiya.tech/api");
-const exportService = new ExportService("https://dev.tabiya.tech/api");
+const importDirectorService = new ImportDirectorService(getApiUrl());
+const modelInfoService = new ModelInfoService();
+const exportService = new ExportService(getApiUrl());
```
Comment (irumvanselme) [inline frontend/src/modeldirectory/ModelDirectory.tsx:32]: why specifically modelInfoService was the constructor changed?

File: frontend/public/data/env.js (hunk 1 above)
Comment (rav3n11) [reply frontend/public/data/env.js:3]: This was a mistake. not sure how I managed to commit the file. I'll push with force and remove it

File: frontend/src/modelInfo/modelInfo.service.ts (hunk 2 above)
Comment (rav3n11) [reply frontend/src/modelInfo/modelInfo.service.ts:42]: Yeah. But for this I was thinking that our services dont really need to have the APIserverUrl passed into them since there is a service for it. They can simply care about the route and we can simplify their implementation a bit. But this does need some discussion for sure.

File: .github/workflows/build-frontend.yml (hunk 3 above)
Comment (rav3n11) [reply .github/workflows/build-frontend.yml:46]: I agree. I'll add that documentation as part of this branch as well.

File: frontend/src/modeldirectory/ModelDirectory.tsx (hunk 4 above)
Comment (rav3n11) [reply frontend/src/modeldirectory/ModelDirectory.tsx:32]: Yeah.. I wanted some feedback on that before I add the others.
